src/scripts/main.js

```diff
@@ -1,3 +1,206 @@
+    }
+
+    if (columnIndex === 4) {
+      cellA = parseFloat(cellA.slice(1).split(',').join(''));
+      cellB = parseFloat(cellB.slice(1).split(',').join(''));
+    }
+
+    if (sortDirection) {
+      return cellA > cellB ? 1 : -1;
+    }
+
+    return cellA > cellB ? -1 : 1;
+  });
+
+  rows.forEach((row) => table.querySelector('tbody').appendChild(row));
+  sortDirection = !sortDirection;
+}
+
+table.querySelectorAll('th').forEach((header, index) => {
+  header.addEventListener('click', () => sortTable(index));
+});
+
+// 2. When user clicks on a row, it should become selected.
+let activeRow = null;
+
+table.querySelectorAll('tbody tr').forEach((row) => {
+  row.addEventListener('click', () => {
+    if (activeRow) {
+      activeRow.classList.remove('active');
+    }
+
+    row.classList.add('active');
+    activeRow = row;
+  });
+});
+
+// 3. Write a script to add a form to the document.
+// Form allows users to add new employees to the spreadsheet.
+const formHtml = `
+  <form class="new-employee-form">
+    <label>
+      Name:
+      <input
+        name="name"
+        type="text"
+        data-qa="name"
+      />
+    </label>
+    <label>
+      Position:
+      <input
+        name="position"
+        type="text"
+        data-qa="position"
+      />
+    </label>
+    <label>
+      Office:
+      <select
+        name="office"
+        data-qa="office"
+      >
+        <option value="Tokyo">Tokyo</option>
+        <option value="Singapore">Singapore</option>
+        <option value="London">London</option>
+        <option value="New York">New York</option>
+        <option value="Edinburgh">Edinburgh</option>
+        <option value="San Francisco">San Francisco</option>
+      </select>
+    </label>
+    <label>
+      Age:
+      <input
+        name="age"
+        type="number"
+        data-qa="age"
+      />
+    </label>
+    <label>
+      Salary:
+      <input
+        name="salary"
+        type="number"
+        data-qa="salary"
+      />
+    </label>
+    <button
+      type="button"
+      id="save-button"
+    >
+      Save to table
+    </button>
+  </form>
+  `;
+
+table.insertAdjacentHTML('afterend', formHtml);
+
+// 4. Show notification if form data is invalid
+function pushNotification(message, type) {
+  const notification = document.createElement('div');
+  const titleSpan = document.createElement('span');
+
+  notification.className = `notification ${type}`;
+  notification.setAttribute('data-qa', 'notification');
+
+  titleSpan.className = 'title';
+  titleSpan.textContent = type.toUpperCase();
+
+  notification.appendChild(titleSpan);
+  notification.innerHTML += ` <span>${message}</span>`;
+  document.body.appendChild(notification);
+
+  setTimeout(() => {
+    notification.style.display = 'none';
+  }, 2000);
+}
+
+function validateFormData(formData) {
+  if (formData.name.length < 4) {
+    pushNotification('Name must be at least 4 characters long.', 'error');
+
+    return false;
+  }
```

Comment:
The validation checks in the validateFormData function could be more strict. For example, it only checks if the name length is less than 4, but it does not check if the name contains only letters and spaces. Additionally, it does not check if the position, office, and salary fields contain valid values.